File: feign-core/src/main/java/name/abhijitsarkar/feign/core/MatchersAutoConfiguration.java
/*
 * Copyright (c) 2016, the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * License for more details.
 */

package name.abhijitsarkar.feign.core;

import name.abhijitsarkar.feign.core.matcher.DefaultBodyMatcher;
import name.abhijitsarkar.feign.core.matcher.DefaultHeadersMatcher;
import name.abhijitsarkar.feign.core.matcher.DefaultMethodMatcher;
import name.abhijitsarkar.feign.core.matcher.DefaultPathMatcher;
import name.abhijitsarkar.feign.core.matcher.DefaultQueriesMatcher;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * @author Abhijit Sarkar
 */
@Configuration
@ConditionalOnProperty(prefix = "feign.matchers", name = "disable",
        havingValue = "false", matchIfMissing = true)
public class MatchersAutoConfiguration {
    @Bean
    DefaultPathMatcher defaultPathMatcher() {
        return new DefaultPathMatcher();
    }

    @Bean
    DefaultMethodMatcher defaultMethodMatcher() {
        return new DefaultMethodMatcher();
    }

    @Bean
    DefaultQueriesMatcher defaultQueriesMatcher() {
        return new DefaultQueriesMatcher();
    }

    @Bean
    DefaultHeadersMatcher defaultHeadersMatcher() {
        return new DefaultHeadersMatcher();
    }

    @Bean
    DefaultBodyMatcher defaultBodyMatcher() {
        return new DefaultBodyMatcher();
    }
}

File: feign-core/src/main/scala/name/abhijitsarkar/feign/core/MatchersAutoConfiguration.scala
/*
 * Copyright (c) 2016, the original author or authors.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; version 3 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * A copy of the GNU General Public License accompanies this software,
 * and is also available at http://www.gnu.org/licenses.
 */

package name.abhijitsarkar.feign.core

import name.abhijitsarkar.feign.core.matcher._
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.context.annotation.{Bean, Configuration}

/**
  * @author Abhijit Sarkar
  */
@Configuration
@ConditionalOnProperty(prefix = "feign.matchers", name = Array("disable"), havingValue = "false", matchIfMissing = true)
class MatchersAutoConfiguration {
  @Bean def defaultPathMatcher = new DefaultPathMatcher

  @Bean def defaultMethodMatcher = new DefaultMethodMatcher

  @Bean def defaultQueriesMatcher = new DefaultQueriesMatcher

  @Bean def defaultHeadersMatcher = new DefaultHeadersMatcher

  @Bean def defaultBodyMatcher = new DefaultBodyMatcher
}
